### PR TITLE
Do not prepackage charts

### PR DESCRIPTION
**Pull request by JakobGray**
Issue: https://github.com/stolostron/backlog/issues/21972

Uses CHART_VERSION environment variable to package charts when the server starts up and serves them in the same directory. Requires https://github.com/stolostron/multiclusterhub-operator/pull/797 in order to get variable injected into the deployment.

For reviewers: Files changed in the pkg directory are what changed. Everything in the multiclusterhub/charts directory is from moving the tgzs to their unpackaged form.

### CODE REVIEW

File: multiclusterhub/charts/application-chart/templates/applicationconsole-service.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ports:
+    - port: 3001
+      targetPort: 3001
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ template "application.name" . }}
+    component: "applicationui"
+    release: {{ .Release.Name }}
+
```
Comment (ray-harris) [inline multiclusterhub/charts/application-chart/templates/applicationconsole-service.yaml:31]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/cluster-backup-chart/templates/oadp-operators-group.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+kind: OperatorGroup
+metadata:
+  name: {{ .Values.oadpOperator.name }}-group
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-2"
+spec:
+  targetNamespaces:
+  - {{ .Release.Namespace }}
```
Comment (ray-harris) [inline multiclusterhub/charts/cluster-backup-chart/templates/oadp-operators-group.yaml:15]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/cluster-lifecycle/templates/clustermanagementaddon-clusterrolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ .Values.org }}:{{ .Release.Name }}:clustermanagementaddons-readonly-v2
+subjects:
+- kind: Group
+  name: system:authenticated
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: {{ .Values.org }}:{{ .Release.Name }}:clustermanagementaddons-readonly-v2
+  apiGroup: rbac.authorization.k8s.io
```
Comment (ray-harris) [inline multiclusterhub/charts/cluster-lifecycle/templates/clustermanagementaddon-clusterrolebinding.yaml:14]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/grc/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+apiVersion: v1
+appVersion: "1.0.0"
+description:  helm chart for multicloud grc
+name: grc
+version: 2.1.0
+category: "Development"
+verified: "RHACM"
+keywords:
+  - acm
+  - grc
```
Comment (ray-harris) [inline multiclusterhub/charts/grc/Chart.yaml:12]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/grc/templates/grc-clusterrolebinding.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    app.kubernetes.io/name: {{ template "grc.name" . }}
+    helm.sh/chart: {{ template "grc.chart" . }}
+roleRef:
+  kind: ClusterRole
+  name: {{ .Values.org }}:{{ .Release.Name }}:clusterrole
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: grc-sa
+    namespace: {{ .Release.Namespace }}
```
Comment (ray-harris) [inline multiclusterhub/charts/grc/templates/grc-clusterrolebinding.yaml:25]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/grc/templates/grc-rolebinding.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    app.kubernetes.io/name: {{ template "grc.name" . }}
+    helm.sh/chart: {{ template "grc.chart" . }}
+roleRef:
+  kind: Role
+  name: {{ .Release.Name }}-grc-role
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: grc-sa
+
```
Comment (ray-harris) [inline multiclusterhub/charts/grc/templates/grc-rolebinding.yaml:24]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/insights-chart/templates/insights-client-service.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  ports:
+    - port: 3030
+      targetPort: 3030
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ template "insights.name" . }}
+    component: "insights-client"
+    release: {{ .Release.Name }}
+  sessionAffinity: ClientIP
```
Comment (ray-harris) [inline multiclusterhub/charts/insights-chart/templates/insights-client-service.yaml:26]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/insights-chart/templates/policyreport-servicemonitor.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    tlsConfig:
+      insecureSkipVerify: true
+  jobLabel: insights-app
+  selector:
+    matchLabels:
+      insights-app: {{ .Values.insights.name }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
```
Comment (ray-harris) [inline multiclusterhub/charts/insights-chart/templates/policyreport-servicemonitor.yaml:34]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/management-ingress/templates/management-ingress-clusterrole.yaml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  - events
+  verbs:
+  - create
+  - patch
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - ingresses/status
+  verbs:
+  - update
```
Comment (ray-harris) [inline multiclusterhub/charts/management-ingress/templates/management-ingress-clusterrole.yaml:75]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/management-ingress/templates/management-ingress-clusterrolebinding.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    app.kubernetes.io/name: {{ .Release.Name }}
+    helm.sh/chart: {{ .Chart.Name }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Values.org }}:{{ .Release.Name }}:clusterrole
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Release.Name }}-sa
+    namespace: {{ .Release.Namespace }}
```
Comment (ray-harris) [inline multiclusterhub/charts/management-ingress/templates/management-ingress-clusterrolebinding.yaml:23]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/search-prod/templates/aggregator-configmap.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    config: acm-proxyserver
+data:
+  service: "{{ .Release.Namespace }}/search-aggregator"
+  port: "3010"
+  path: "/aggregator/clusters/"
+  sub-resource: "/sync"
+  use-id: "true"
+  secret: "{{ .Release.Namespace }}/search-aggregator-certs"
+  caConfigMap: search-ca-crt
```
Comment (ray-harris) [inline multiclusterhub/charts/search-prod/templates/aggregator-configmap.yaml:17]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/search-prod/templates/aggregator-service-account.yaml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2020 Red Hat, Inc.
+# Copyright Contributors to the Open Cluster Management project
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: search-aggregator
```
Comment (ray-harris) [inline multiclusterhub/charts/search-prod/templates/aggregator-service-account.yaml:7]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/search-prod/templates/api-role-binding.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: {{ .Values.org }}:{{ .Release.Name }}:search-api
+subjects:
+- kind: ServiceAccount
+  name: search-api
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ .Values.org }}:{{ .Release.Name }}:search-api
+  apiGroup: rbac.authorization.k8s.io
```
Comment (ray-harris) [inline multiclusterhub/charts/search-prod/templates/api-role-binding.yaml:15]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/search-prod/templates/api-service.yaml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+  ports:
+    - port: 4010
+      targetPort: 4010
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ .Values.search.name }}
+    component: "search-api"
+    release: {{ .Release.Name }}
+  sessionAffinity: ClientIP
```
Comment (ray-harris) [inline multiclusterhub/charts/search-prod/templates/api-service.yaml:29]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/search-prod/templates/cm-search-redis-ca-crt.yaml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    service.beta.openshift.io/inject-cabundle: "true"
+  name: search-ca-crt
```
Comment (ray-harris) [inline multiclusterhub/charts/search-prod/templates/cm-search-redis-ca-crt.yaml:6]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/volsync-addon-controller/templates/_helpers.tpl [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "volsync-addon-controller.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
```
Comment (ray-harris) [inline multiclusterhub/charts/volsync-addon-controller/templates/_helpers.tpl:32]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-clusterrolebinding.yaml [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    helm.sh/chart: {{ template "volsync-addon-controller.chart" . }}
+    component: clusterrolebinding
+subjects:
+- kind: ServiceAccount
+  name: {{ template "volsync-addon-controller.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ .Values.org }}:{{ .Release.Name }}:clusterrole
+  apiGroup: rbac.authorization.k8s.io
```
Comment (ray-harris) [inline multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-clusterrolebinding.yaml:22]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-deployment.yaml [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+            value: {{ .Values.hubconfig.proxyConfigs.NO_PROXY }}
+          {{- end }}
+      {{- if .Values.pullSecret }}
+      imagePullSecrets:
+      - name: {{ .Values.pullSecret }}
+      {{- end }}
+      {{- with .Values.hubconfig.nodeSelector }}
+      nodeSelector:
+      {{ toYaml . | indent 8 }}
+      {{- end }}
```
Comment (ray-harris) [inline multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-deployment.yaml:141]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

File: multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-serviceaccount.yaml [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+  labels:
+    app: {{ template "volsync-addon-controller.name" . }}
+    chart: {{ template "volsync-addon-controller.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/name: {{ template "volsync-addon-controller.name" . }}
+    helm.sh/chart: {{ template "volsync-addon-controller.chart" . }}
+    component: volsync-addon-controller-serviceaccount
```
Comment (ray-harris) [inline multiclusterhub/charts/volsync-addon-controller/templates/volsync-addon-controller-serviceaccount.yaml:14]: Missing newline at end of last line.

https://stackoverflow.com/a/729795/1437822

**Review by ray-harris — COMMENTED** [review]
Several files are missing '\n' at the end of the last line. If these are being copied from elsewhere, no need to fix here, but we should be sure when changing or adding text files, that we have newlines at the end of the last line.

https://stackoverflow.com/a/729795/1437822

**Comment by JakobGray** [comment]
> Several files are missing '\n' at the end of the last line. If these are being copied from elsewhere, no need to fix here, but we should be sure when changing or adding text files, that we have newlines at the end of the last line.
> 
> https://stackoverflow.com/a/729795/1437822

Yeah those are all from the charts so if we change them it would have to be done there. Nothing to do here.

**Pull request merged**